definitions/procedures/puppet/enable_puppet.rb

```diff
@@ -0,0 +1,22 @@
+module Procedures::Puppet
+  class EnablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Enable Puppet feature'
+    end
+
+    def run
+      feature(:installer).run(installer_arguments_enabling_puppet.join(' '), :interactive => false)
+    end
+
+    private
+
+    def installer_arguments_enabling_puppet
+      [
+        '--enable-puppet',
```

Comment:
For vanilla foreman you do want the server enabled it is simply:
```suggestion
        '--enable-puppet --puppet-server true',
```

Katello additionally needs you need to specify the certificates. See https://github.com/theforeman/foreman-documentation/pull/799 or https://github.com/theforeman/forklift/blob/64ae4e81d0ae2cad13a4c925b0b4ccb0c923057e/bats/fb-test-puppet.bats#L18 for the horrible mess.